File: lib/hatena.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Text::Hatena;
# use File::Slurp;

# my $filename = shift or die "usage: $0 [hatena style text file name]\n";
# my $text = read_file($filename);
my $text = shift or die "usage: $0 [hatena style text file name]\n";
my $html = Text::Hatena->parse($text);
$html =~ s#</?div.*?>.*?\n##gi;
print $html;

File: lib/hatena.rb
require 'rubygems'
require 'hparser'

#unless filename = ARGV.shift
#      puts "usage: #{$0} [text file]"
#        exit
#end

# text = File.open(filename).read
text = ARGV.shift


parser = HParser::Parser.new
hatena = parser.parse text

puts hatena.map{|e| e.to_html}
